### ISSUE TITLE
Equipment: extra field for amount of items

### ISSUE THREAD

**Issue by Medsmiley** [elyukai/optolith-client#1119]
# 1 Basic Idea
## Shop
- just like with the costumizable AP boxes there could be a box just left to the add-button
- default entry is 1 meaning one item to add
- if you want to add multiple items of the same kind at once, you can just change that number

## Inventory
- in the list just left to the name column there could be another column containing the number of items you have
- ideally you could just click there and change the amount again
- or if there are reasons against it: just change the amount like today via the edit button

## Sorting
On top of all this there should be an additional possibility to sort your items regarding the amount you have of them.

## Viewability
- there was an issue with adding items to often by accident earlier which lead to the list being filtered to not show items you already have in the inventory
- with the above solution where you can choose how many items you want, you don't need to be able to click one item multiple times in a row
- therefor there could be a small time unit in which you can't click on the same add button again -> so you can't accidentally add something more than once
- then the list can still show things you already added -> and if you decide that you want something more than you thought 2 seconds before, you just add more from the same screen without having to leave it, find the item in your inventory, editing and then going back to the shop and find where you left

# 2 A slightly other idea
- consider the same like above, but:
- the small box already shows the total amount of items you own
- so instead of a button to add something, you just have the field with a default of 0 or an empty box that in special cases has a grey number in it, if the recommended amount of an item is !=1
- and if you want to add smth to your inventory, you can either
  - click in the box and type a number
  - click on a `+` button (which would have to be there then, just like a `-` button, too. Best if they are right/left of the box)
In conclusion
- the shop kind of already shows the inventory (very useful)
- the inventory _only shows the possessions_ and additional information like price, weight, place etc
- the inventory shows costum items for which there is no template in the game
